NineChronicles.Headless/GraphTypes/ActionQueryFields/HackAndSlash.cs

```diff
@@ -0,0 +1,102 @@
+                {
+                    Name = "costumeIds",
+                    Description = "List of costume id for equip."
+                },
+                new QueryArgument<ListGraphType<GuidGraphType>>
+                {
+                    Name = "equipmentIds",
+                    Description = "List of equipment id for equip."
+                },
+                new QueryArgument<ListGraphType<GuidGraphType>>
+                {
+                    Name = "consumableIds",
+                    Description = "List of consumable id for use."
+                },
+                new QueryArgument<ListGraphType<NonNullGraphType<RuneSlotInfoInputType>>>
+                {
+                    Name = "runeSlotInfos",
+                    DefaultValue = new List<RuneSlotInfo>(),
+                    Description = "List of rune slot info for equip."
+                },
+                new QueryArgument<IntGraphType>
+                {
+                    Name = "stageBuffId",
+                    Description = "Buff ID for this stage"
+                }
+            ),
+            resolve: context =>
+            {
+                try
+                {
+                    Address avatarAddress = context.GetArgument<Address>("avatarAddress");
+                    int worldId = context.GetArgument<int>("worldId");
+                    int stageId = context.GetArgument<int>("stageId");
+                    List<Guid> costumeIds = context.GetArgument<List<Guid>?>("costumeIds") ?? new List<Guid>();
+                    List<Guid> equipmentIds = context.GetArgument<List<Guid>?>("equipmentIds") ?? new List<Guid>();
+                    List<Guid> consumableIds =
+                        context.GetArgument<List<Guid>?>("consumableIds") ?? new List<Guid>();
+                    List<RuneSlotInfo> runeSlotInfos =
+                        context.GetArgument<List<RuneSlotInfo>?>("runeSlotInfos") ?? new List<RuneSlotInfo>();
+                    int? stageBuffId = context.GetArgument<int?>("stageBuffId");
+
+                    if (!(standaloneContext.BlockChain is { } chain))
+                    {
+                        throw new InvalidOperationException("BlockChain not found in the context");
+                    }
+
+                    NCAction action = new HackAndSlash
+                    {
+                        AvatarAddress = avatarAddress,
+                        WorldId = worldId,
+                        StageId = stageId,
+                        Costumes = costumeIds,
+                        Equipments = equipmentIds,
+                        Foods = consumableIds,
+                        RuneInfos = runeSlotInfos,
+                        StageBuffId = stageBuffId
+                    };
+                    return Encode(context, action);
+                }
+                catch (Exception e)
+                {
+                    var msg = $"Unexpected exception occurred during {typeof(ActionMutation)}: {e}";
```

Comment:
Could you tell me why the exception message contains the type of `ActionMutation`?

Reply:
Oh, I just copied that `catch` block from `ActionMutation.cs` and forgot to change name to right one
Thanks for pointing out!